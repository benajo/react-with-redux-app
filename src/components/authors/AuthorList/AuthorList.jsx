import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AuthorList = ({ authors, onDeleteClick }) => (authors.length > 0 ? (
  <table className="table">
    <thead>
      <tr>
        <th>Name</th>
        <th>&nbsp;</th>
      </tr>
    </thead>
    <tbody>
      {authors.map((author) => (
        <tr key={author.id}>
          <td>
            <Link to={`/author/${author.slug}`}>{author.name}</Link>
          </td>
          <td>
            <button
              type="submit"
              className="btn btn-outline-danger"
              onClick={() => onDeleteClick(author)}
            >
              Delete
            </button>
          </td>
        </tr>
      ))}
    </tbody>
  </table>
) : (
  <p>There are no authors to display.</p>
));

AuthorList.propTypes = {
  authors: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  })).isRequired,
  onDeleteClick: PropTypes.func.isRequired,
};

export default AuthorList;
