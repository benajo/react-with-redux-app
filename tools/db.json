{
  "courses": [
    {
      "id": 1,
      "title": "Securing React Apps with Auth0.1",
      "slug": "react-auth0-authentication-security",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 2,
      "title": "React: The Big Picture",
      "slug": "react-big-picture",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 3,
      "title": "Creating Reusable React Components",
      "slug": "react-creating-reusable-components",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 4,
      "title": "Building a JavaScript Development Environment",
      "slug": "javascript-development-environment",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 5,
      "title": "Building Applications with React and Redux",
      "slug": "react-redux-react-router-es6",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 6,
      "title": "Building Applications in React and Flux",
      "slug": "react-flux-building-applications",
      "authorId": 1,
      "category": "JavaScript"
    },
    {
      "id": 7,
      "title": "Clean Code: Writing Code for Humans",
      "slug": "writing-clean-code-humans",
      "authorId": 1,
      "category": "Software Practices"
    },
    {
      "id": 8,
      "title": "Architecting Applications for the Real World",
      "slug": "architecting-applications-dotnet",
      "authorId": 1,
      "category": "Software Architecture"
    },
    {
      "id": 9,
      "title": "Becoming an Outlier: Reprogramming the Developer Mind",
      "slug": "career-reboot-for-developer-mind",
      "authorId": 1,
      "category": "Career"
    },
    {
      "id": 10,
      "title": "Web Component Fundamentals",
      "slug": "web-components-shadow-dom",
      "authorId": 1,
      "category": "HTML5"
    },
    {
      "id": 11,
      "title": "test",
      "authorId": 1,
      "category": "testcat",
      "createdAt": 1569005636631,
      "slug": "test"
    },
    {
      "id": 12,
      "title": "test2",
      "authorId": 3,
      "category": "asd",
      "createdAt": 1569005961845,
      "slug": "test2"
    },
    {
      "id": 13,
      "title": "jjj",
      "authorId": 3,
      "category": "jjjjkk",
      "createdAt": 1569006187120,
      "slug": "jjj"
    },
    {
      "id": 14,
      "title": "hello",
      "authorId": 1,
      "category": "you",
      "createdAt": 1569006296726,
      "slug": "hello"
    },
    {
      "id": 15,
      "title": "foo",
      "authorId": 1,
      "category": "bar",
      "createdAt": 1569006310621,
      "slug": "foo"
    },
    {
      "id": 16,
      "title": "foo",
      "authorId": 1,
      "category": "bar",
      "createdAt": 1569006318445,
      "slug": "foo"
    }
  ],
  "authors": [
    {
      "id": 1,
      "name": "Cory House"
    },
    {
      "id": 2,
      "name": "Scott Allen"
    },
    {
      "id": 3,
      "name": "Dan Wahlin"
    }
  ]
}