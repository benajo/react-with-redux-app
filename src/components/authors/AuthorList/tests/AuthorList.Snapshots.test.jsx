import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import AuthorList from '../AuthorList';
import { authors } from '../../../../../tools/mockData';

it('render list of authors', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <AuthorList
        authors={authors}
        onDeleteClick={jest.fn()}
      />
    </MemoryRouter>,
  );

  expect(tree).toMatchSnapshot();
});

it('render empty list of authors', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <AuthorList
        authors={[]}
        onDeleteClick={jest.fn()}
      />
    </MemoryRouter>,
  );

  expect(tree).toMatchSnapshot();
});
