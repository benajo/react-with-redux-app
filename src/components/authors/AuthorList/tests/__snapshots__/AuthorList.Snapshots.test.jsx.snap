// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty list of authors 1`] = `
<p>
  There are no authors to display.
</p>
`;

exports[`render list of authors 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Â 
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a
          href="/author/cory-house"
          onClick={[Function]}
        >
          Cory House
        </a>
      </td>
      <td>
        <button
          className="btn btn-outline-danger float-right"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/author/scott-allen"
          onClick={[Function]}
        >
          Scott Allen
        </a>
      </td>
      <td>
        <button
          className="btn btn-outline-danger float-right"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/author/dan-wahlin"
          onClick={[Function]}
        >
          Dan Wahlin
        </a>
      </td>
      <td>
        <button
          className="btn btn-outline-danger float-right"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
